app/src/main/java/com/example/lastfm/MainActivity.java

```diff
@@ -4,81 +4,128 @@
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
+import android.icu.text.Edits;
+import android.os.AsyncTask;
 import android.os.Bundle;
-import android.widget.ImageView;
 import android.widget.TextView;
 
-import com.squareup.picasso.Picasso;
-
-import com.example.lastfm.NewsData;
-import com.example.lastfm.NewsAdapter;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
 
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
 
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
 import java.util.Arrays;
 import java.util.Collection;
 
 public class MainActivity extends AppCompatActivity {
-    private ImageView newsImage;
-    private TextView textHeader;
-    private TextView textDescription;
 
 
-    private RecyclerView newsRecyclerView;
-    private NewsAdapter newsAdapter;
 
+    private static final String APP_ID = "b4ab3bf82dcb495e182e04cfc1f12b7b";
```

Comment:
Why do u need this variable? Why is it static?

Reply:
Changed

Reply:
Needed to set APP_ID to get request

Reply:
move this variable to the resource file